File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task61_PERPLEXITY_llama-3-sonar-large-32k-chat.java
import javax.xml.parsers.DocumentBuilderFactory;
import javax.xml.parsers.DocumentBuilder;
import org.w3c.dom.Document;
import org.w3c.dom.Element;

public class Main {
    public static void main(String[] args) throws Exception {
        String xmlString = "<root><person><name>John</name><age>30</age></person></root>";
        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
        DocumentBuilder builder = factory.newDocumentBuilder();
        Document doc = builder.parse(new java.io.ByteArrayInputStream(xmlString.getBytes()));
        Element rootElement = doc.getDocumentElement();
        System.out.println("Root Element: " + rootElement.getTagName());
    }
}

File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task61_PERPLEXITY_llama-3-sonar-large-32k-chat.cpp
// C++ does not have a built-in XML parsing library, so we'll use the Xerces-C++ library
// This code assumes you have the Xerces-C++ library installed and linked

#include <xercesc/parsers/XercesDOMParser.hpp>
#include <xercesc/dom/DOM.hpp>
#include <xercesc/util/PlatformUtils.hpp>

#include <iostream>

int main() {
    try {
        xercesc::XMLPlatformUtils::Initialize();
        XercesDOMParser* parser = new XercesDOMParser();
        parser->parse("xml_string.xml"); // replace with your XML string
        xercesc::DOMDocument* doc = parser->adoptDocument();
        xercesc::DOMElement* root = doc->getDocumentElement();
        std::cout << root->getTagName() << std::endl;
        delete parser;
        xercesc::XMLPlatformUtils::Terminate();
    } catch (const xercesc::XMLException& e) {
        std::cerr << "Error: " << e.getMessage() << std::endl;
        return 1;
    }
    return 0;
}